src/formElementPlugins/dropdownChoice/src/ApplicationView.tsx

```diff
@@ -21,53 +22,93 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
     search,
     options,
     optionsDisplayProperty,
+    hasOther,
     default: defaultOption,
   } = parameters
 
-  const [selectedIndex, setSelectedIndex] = useState<number>()
+  const [selectedIndex, setSelectedIndex] = useState<any>()
+  const [localOptions, setLocalOptions] = useState<any[]>([])
+  //saving selected value in local state, so that when options are re rendered, we retain this value
+  const [addedOptions, setAddedOptions] = useState<any[]>([])
```

Comment:
We should create a proper `type` for this and not use `any`.

Reply:
Also, you'll need to populate the initial value. Currently when saving an additional item and reloading, the "added option" no longer shows up.

After "options" as loaded, if the "currentReponse" text is not included in "options", then you'll need to add it to "addedOptions" at that point.